File: 122_BestTimeToBuyAndSellStock_II.py
###51.7%
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        if not prices:
            return 0
        maxP = 0
        i, size = 1, len(prices)
        while i != size:
            curr = prices[i] - prices[i - 1]
            if curr > 0:
                maxP += curr
            i += 1
        return maxP

##A good solution from LeetCode		
class Solution(object):
    def maxProfit(self, prices):
        return sum(max(prices[i + 1] - prices[i], 0) for i in range(len(prices) - 1))

File: 122_BestTimeToBuyAndSellStock_II.cpp
//38.83%
class Solution {
public:
    int maxProfit(vector<int>& prices) {
		if(prices.empty())
			return 0;
        int currDiff, maxP(0);
		for(int i = 1; i < prices.size(); ++i){
			currDiff = max(0, prices[i] - prices[i - 1]);
			maxP += max(0, currDiff);
		}
		return maxP;
    }
};